src/App.tsx

```diff
@@ -15,71 +16,92 @@ export const goods = [
-    </h1>
+  selectedElement = (element: string) => {
+    this.setState({ selectedGood: element });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  removedElement = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        { !selectedGood
+          ? (
+            <h1 className="title">
+              No goods selected
+            </h1>
+          )
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+          : (
+            <h1 className="title is-flex is-align-items-center">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              {`${selectedGood} is selected`}
+
+              <button
+                data-cy="ClearButton"
+                type="button"
+                aria-label="ClearButton"
+                className="delete ml-3"
+                onClick={() => this.setState(
+                  this.removedElement,
+                )}
+              />
+            </h1>
+          )}
+
+        <table className="table">
+          <tbody>
+            {goods.map(element => (
+              <tr
+                data-cy="Good"
+                className={selectedGood === element
+                  ? 'has-background-success-light'
+                  : ''}
+              >
+                <td>
+                  <button
+                    data-cy={selectedGood === element
+                      ? 'RemoveButton'
+                      : 'AddButton'}
+                    type="button"
+                    className={
+                      classNames('button',
+                        { 'is-info': selectedGood === element })
+                    }
+                    onClick={() => (
+                      selectedGood === element
+                        ? this.removedElement()
+                        : this.selectedElement(element)
+                    )}
```

Comment:
I believe that you can create a separate method that receives element and or remove selected good or update it

Reply:
thanks for faith)